DonjinKrawler-Client/src/main/java/donjinkrawler/adapter/Mp3Player.java

```diff
@@ -1,32 +1,41 @@
 package donjinkrawler.adapter;
 
-import javax.sound.sampled.AudioInputStream;
-import javax.sound.sampled.AudioSystem;
-import javax.sound.sampled.Clip;
+import javax.sound.sampled.*;
 import java.io.File;
 
 public class Mp3Player implements AdvancedMediaPlayer {
 
     private Clip clip;
 
     @Override
-    public void playMp3(String fileName) {
+    public void playMp3(String fileName, boolean repeat) {
         try {
             File soundFile = new File(ClassLoader.getSystemResource("sounds/".concat(fileName)).getFile());
             AudioInputStream audioInput = AudioSystem.getAudioInputStream(soundFile);
             clip = AudioSystem.getClip();
             clip.open(audioInput);
+
+            clip.addLineListener(new LineListener() {
+                @Override
```

Comment:
"Avoid duplication of code"
Can we extract this for both players?

Reply:
No.